app/common/services/icon-manager.js

```diff
@@ -0,0 +1,42 @@
+module.exports = [
+    'FontAwesomeIcons',
+    'Util',
+    '_',
+    'Notify',
+function (
+    FontAwesomeIcons,
+    Util,
+    _,
+    Notify
+) {
+
+    var iconSets = {
+        fontAwesome: FontAwesomeIcons.getIconSet(),
+        noun: {}
+    };
+
+    var IconManager = {
+
+        getIconSetArray: function (iconLibName) {
+          var iconSetArray = [];
+          var iconLibNames = iconLibName.split(' ');
+
+          // @todo: simplify this to reduce loops
+          // push array construction to the endpoint
+          // and/or store values with prefix already attached
+          _.each(iconLibNames, function (libName) {
```

Comment:
I'd like to see something similar.. but as long as this works lets land it and move on.